swap_meet/vendor.py

```diff
@@ -1,2 +1,94 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    '''
+    Vendor class has an attribute named inventory which starts as an empty list by default and complete 
+    methods such as adding, removing, swapping, and figuring out best categories in your inventory and 
+    your friend's inventory.
+    '''
+    
+    def __init__(self, inventory = None):
+        '''
+        Attribute named inventory which starts as an empty list by default, but can be pass in a list.
+        '''
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    
+    def add(self, add_item):
+        '''
+        Instance method that takes in an item and add it to the inventory list and return the item that was added.
+        '''
+        self.inventory.append(add_item)
+        return add_item
+
+    def remove(self, remove_item):
+        '''
+        Instance method that removes an item from the inventory list and return the item that was removed.
+        '''
+        if remove_item in self.inventory:
+            self.inventory.remove(remove_item)
+            return remove_item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        '''
+        Instance method that takes in a category string and returns a list of items in the inventory with that category.
+        '''
+        category_list = []
+        for item in self.inventory:
+            if category == item.category:
+                category_list.append(item)
+        return category_list
```

Comment:
Nice, nice. This method is a great candidate for using list comprehension if you want to refactor it since our conditional statement we check is pretty simple.

General syntax for list comp:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:

```python
category_list = [item for item in self.inventory if item.category == category]
```

You can also forgo saving the list to the variable category_items and do something like:
```python 
def get_by_category(self, category):
    return [item for item in self.inventory if item.category == category]
```